diff -up ibus-fbterm-0.9.1/ibus-fbterm-launch.orig ibus-fbterm-0.9.1/ibus-fbterm-launch
--- ibus-fbterm-0.9.1/ibus-fbterm-launch.orig	2010-01-22 16:29:03.542929160 +1000
+++ ibus-fbterm-0.9.1/ibus-fbterm-launch	2010-01-22 16:29:43.456734687 +1000
@@ -12,7 +12,7 @@ if [ ${#TTY} -eq ${#TTY_NUM} ]; then
 	exit 1
 fi
 
-eval `dbus-launch --auto-syntax`
+eval `dbus-launch --sh-syntax`
 
 export DISPLAY=:9${TTY_NUM}.0
 
