--- ibus-fbterm-0.9.1/src/display.c.orig	2012-03-12 16:21:06.582175770 +0900
+++ ibus-fbterm-0.9.1/src/display.c	2012-03-12 16:32:37.057787019 +0900
@@ -212,7 +212,11 @@ static void calculate_status_win()
 		IBusProperty *prop = ibus_prop_list_get(property_list, i);
 		if (!prop) break;
 
+#if IBUS_CHECK_VERSION (1, 4, 0)
+		w += text_width(ibus_property_get_label(prop)->text);
+#else
 		w += text_width(prop->label->text);
+#endif
 	}
 
 	status_bar_win.x = cursor_x;
@@ -238,8 +242,16 @@ static void draw_status_bar()
 		IBusProperty *prop = ibus_prop_list_get(property_list, i);
 		if (!prop) break;
 
+#if IBUS_CHECK_VERSION (1, 4, 0)
+                IBusText *text = ibus_property_get_label(prop);
+		draw_text(x, y, COLOR_FG, COLOR_BG,
+                          text->text,
+                          strlen(text->text));
+		x += FW(text_width(text->text));
+#else
 		draw_text(x, y, COLOR_FG, COLOR_BG, prop->label->text, strlen(prop->label->text));
 		x += FW(text_width(prop->label->text));
+#endif
 
 		char space = ' ';
 		draw_text(x, y, COLOR_FG, COLOR_BG, &space, 1);
--- ibus-fbterm-0.9.1/src/ibus-fbterm.c.orig	2012-03-13 19:47:33.036218019 +0900
+++ ibus-fbterm-0.9.1/src/ibus-fbterm.c	2012-03-13 19:48:13.694187234 +0900
@@ -119,13 +119,17 @@ static void im_active()
 	debug("im active\n");
 	modifier_state = 0;
 	init_keycode_state();
+#if ! IBUS_CHECK_VERSION (1, 4, 99)
 	ibus_input_context_enable(ibus_ctx);
+#endif
 }
 
 static void im_deactive()
 {
 	debug("im deactive\n");
+#if ! IBUS_CHECK_VERSION (1, 4, 99)
 	ibus_input_context_disable(ibus_ctx);
+#endif
 
 	auxiliary_text_win.w = 0;
 	lookup_table_win.w = 0;
